src/lib/user/user.ts

```diff
@@ -1,7 +1,10 @@
-import { Players } from "@rbxts/services";
-import { DefaultUser } from "./types";
+import { Players, RunService } from "@rbxts/services";
 
-export abstract class User implements DefaultUser {
+import { DefaultUserDeclaration, OfflineUserDeclaration } from "./types";
+
+import logger from "../logger";
+
+export class User implements DefaultUserDeclaration {
```

Comment:
Maaaybe this should be abstract? Have load and unload as abstract.

Reply:
It was abstract, though, if it's the initial implementation from where Users are created, it won't be able to actually being constructed from (call `new` on it). 

That way, it interferes with creating users. We could assume that user defines their own User class though, force them to.

Reply:
Or either way, work around a non-abstract class which just extends from it like `DefaultUser extends User`?